pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -643,6 +644,46 @@ func TestPriorityQueue_NominatedPodsForNode(t *testing.T) {
 	}
 }
 
+func TestPriorityQueue_NominatedPodDeleted(t *testing.T) {
+	tests := []struct {
+		name    string
+		podInfo *framework.PodInfo
+		want    bool
+	}{
+		{
+			name:    "alive pod gets added into PodNominator",
+			podInfo: medPriorityPodInfo,
+			want:    true,
+		},
+		{
+			name:    "deleted pod shouldn't be added into PodNominator",
+			podInfo: highPriNominatedPodInfo,
+			want:    false,
+		},
+		{
+			name:    "pod without .status.nominatedPodName specified shouldn't be added into PodNominator",
+			podInfo: highPriorityPodInfo,
+			want:    false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Simulate that highPriNominatedPod gets deleted physically.
+			nominator := NewSafePodNominator(func(pod *v1.Pod) bool {
+				return pod.Name != highPriNominatedPodInfo.Pod.Name
```

Comment:
Can we use the real method here?

Reply:
Maybe not worth it - the real method appears bug-free, doesn't it?

```go
func(pod *v1.Pod) bool {
    _, err := c.informerFactory.Core().V1().Pods().Lister().Pods(pod.Namespace).Get(pod.Name)
    return err == nil
}
```

If you are concerned, I can write a test covering it.

Reply:
Note that there is no unit test providing coverage for the nominator that we use in production. Someone could change the passed function and potentially break users.